src-frontend/app/controllers/procedure.js

```diff
@@ -63,6 +64,31 @@ export default Ember.ObjectController.extend({
             });
 
             this.endPropertyChanges();
+        },
+        generateProtocol: function() {
+            var filename = this.get('model.title') + '.xml';
+
+            Ember.$.ajax({
+                type: 'GET',
+                url: ENV.APP.API_PROCEDURE_URL + this.get('model.id') + ENV.APP.API_GENERATE,
+                beforeSend: function(xhr) {
+                    xhr.setRequestHeader('Authorization', 'Token ' + Cookies.get('authorizationToken'));
+                },
+                success: function(data) {
+                    var blob = new Blob([(new XMLSerializer()).serializeToString(data)], { type: 'text/xml' });
+                    var url = window.URL.createObjectURL(blob);
+
+                    var a = document.createElement("a");
+                    document.body.appendChild(a);
+                    a.setAttribute('id', 'generate-trigger');
+                    a.href = url;
+                    a.download = filename;
+                    a.click();
```

Comment:
It's a "hack" to allow us to save a file from an ajax call. If we can find a way to navigate to api page and have it behave properly, we don't need this.